src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -145,15 +160,14 @@ private String getCookie(URL url){
         return null;
     }
 
-    private boolean sessionHasStarted(InputStream responseStream){
+    private boolean sessionHasStarted(InputStream responseStream) {
         //check the response body
         Gson gson = new Gson();
-        //Map<String,Object> jsonResponse = gson.fromJson(new InputStreamReader(responseStream));
-        JsonObject jsonResponse = gson.fromJson(new InputStreamReader(responseStream), JsonObject.class);
+        JsonObject jsonResponse = gson.fromJson(new InputStreamReader(responseStream), JsonObject
+                .class);
 
         // only check for ok:true, https://issues.apache.org/jira/browse/COUCHDB-1356
         // means we cannot check that the name returned is the one we sent.
-       //return jsonResponse.containsKey("ok") && (Boolean) jsonResponse.get("ok");
         return jsonResponse != null && jsonResponse.has("ok")
                 && !jsonResponse.get("ok").isJsonNull();
```

Comment:
This doesn't check that `ok` has the value `true`, just that it exists, and it isn't `null`